### ISSUE TITLE
Switch is prone to remounts

### ISSUE THREAD

**Issue by wmertens** [ReactTraining/react-router#5591]
See the various issues people have with Switch and keys on Routes causing unintended remounts, like https://github.com/ReactTraining/react-router/issues/4578#issuecomment-325859968 

How about adding an optional prop on Switch, `routes`, which is an array of the Route props? If it is there, `children` would be ignored and it would be Switch that makes the `<Route {...chosenProps}/>` without having to clone the child.

So instead of 

```js
<Switch>
  <Route path="foo" component={Foo} />
  <Route path="bar" component={Bar} />
  <Route path="bar2" component={Bar} />
  <Redirect to="foo" />
</Switch>
```

we'd have

```js
<Switch
  routes={[
    {path:"foo", component: Foo},
    {path:"bar", component: Bar},
    {path:"bar2", component: Bar},
    {redirectTo: "foo"},
  ]}
/>
```

**Comment by timdorr** [comment]
That's just what react-router-config does. So, it already exists.

**Comment by mjackson** [comment]
I agree this is a problem, but I'd like to solve this with components. I'd hate to have 2 ways for people to define routes.

**Comment by timdorr** [comment]
We also have tests and a fix in place in #4592. I'm not sure why this would be happening...

**Comment by wmertens** [comment]
:+1: all makes sense, closing…
